extensions/notification/utils/config.js

```diff
@@ -0,0 +1,89 @@
+const constants = require('../constants');
+
+/**
+ * Transforms array of configurations to a Map
+ * @param  {Object[]} configs - Array of bucket notification configurations
+ * @return {Map} Map of bucket notification configurations
+ */
+function transform(configs) {
+    const configMap = new Map();
+    configs.forEach(config => {
+        if (config.bucket && config.notificationConfiguration) {
+            const bucket = config.bucket;
+            const conf = config.notificationConfiguration.queueConfigurations;
+            if (configMap.has(bucket)) {
+                const val = configMap.get(bucket);
+                configMap.set(bucket, [...val, ...conf]);
+            } else {
+                configMap.set(bucket, [...conf]);
+            }
+        }
+    });
+    return configMap;
+}
+
+/**
+ * Validate object name against bucket notification configuration filters
+ * @param  {Object[]} filterRules - Filter rules from the bucket notification
+ * configuration.
+ * @param  {string} objectName - Name of the object
+ * @return {boolean} true if filter matches the object name
+ */
+function validateFilter(filterRules, objectName) {
+    return filterRules.some(rule => {
+        const name = rule.name;
```

Comment:
If you wish to use shorthands, you can also use the more compact form `const { name, value } = rule;`, but in that case I'm not convinced it helps readability, you could also directly use `rule.name` and `rule.value` to make it explicit.